apps/docs/src/guide/testing/tweaking-the-blockchain.md

```diff
@@ -0,0 +1,23 @@
+# Producing Blocks
+
+You can produce blocks using the `produceBlocks` helper to achieve an arbitrary block height. This is especially useful when you want to do some testing regarding transaction maturity.
+
+<<< @/../../../packages/providers/src/provider.test.ts#Provider-produce-blocks{ts:line-numbers}
+
+# Producing Blocks With Custom Timestamps
+
+You can also produce blocks with custom timestamps using the `produceBlocks` helper by specifying the second optional parameter.
+
+```typescript
+import { fromUnixToTai64, Provider } from 'fuels';
+
+const latestBlockNumber = await provider.produceBlocks(
+  amountOfBlocksToProduce: 3,
+  {
+    blockTimeInterval: '100',
+    startTime: fromUnixToTai64(1620000000), // Use the `fromUnixToTai64` helper to convert unix timestamps to tai64
+  }
+);
+```
```

Comment:
I have gone for an independent code snippet instead of referencing the unit test for this functionality because the unit test is kinda long, and has a lot of code just for assertions:

https://github.com/FuelLabs/fuels-ts/blob/dadc5760549564480b7b580cb5c268030bf7dcd6/packages/providers/src/provider.test.ts#L202-L247

Reply:
The issue with hardcoded snippets is that they can quickly get outdated, and we cannot validate if they are working as expected. The primary motivation behind #894 is to solve these code snippets' clarity x confidence relationship.

From this perspective, could you please write a new test following your rationale, put it inside the apps/docs-snippets package, and then reference it instead?

Reply:
@arboleya @Torres-ssf thanks for pointing this out, and I agree. Please review the test that I have added in place of the hard-coded snippet, and let me know if it looks good!